task/deprecated-image-check/0.1/deprecated-image-check.yaml

```diff
@@ -33,62 +30,66 @@ spec:
+        failure_counter=0
+        error_counter=0
+        export IMAGE_REPO_PATH=$(workspaces.sanity-ws.path)/${IMAGE_REPOSITORY}
+        cat $(results.PYXIS_HTTP_CODE.path) | while read line;
+        do
+          IFS=:' ' read -r http_code IMAGE_REGISTRY IMAGE_REPOSITORY <<< $line; echo "[$http_code] [$IMAGE_REGISTRY] [$IMAGE_REPOSITORY]"
+          if [ "$http_code" == "200" ];
+          then
+            echo "Running conftest using $(params.POLICY_DIR) policy, $(params.POLICY_NAMESPACE) namespace"
+            /usr/bin/conftest test --no-fail ${IMAGE_REPO_PATH}/repository_data.json \
+            --policy $(params.POLICY_DIR) --namespace $(params.POLICY_NAMESPACE) \
+            --output=json 2> ${IMAGE_REPO_PATH}/stderr.txt | tee ${IMAGE_REPO_PATH}/deprecated_image_check_output.json
 
-        if [ "$http_code" == "200" ];
-        then
-          echo "Running conftest using $(params.POLICY_DIR) policy, $(params.POLICY_NAMESPACE) namespace"
-          /usr/bin/conftest test --no-fail $(workspaces.sanity-ws.path)/repository_data.json \
-          --policy $(params.POLICY_DIR) --namespace $(params.POLICY_NAMESPACE) \
-          --output=json 2> $(workspaces.sanity-ws.path)/stderr.txt | tee $(workspaces.sanity-ws.path)/deprecated_image_check_output.json
-          echo "Done!"
-          exit 0
-        elif [ "$http_code" == "404" ];
-        then
-          echo "Registry/image $(params.IMAGE_REGISTRY)/$(params.IMAGE_REPOSITORY) not found in Pyxis" > $(workspaces.sanity-ws.path)/stderr.txt
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        else
-          echo "Unexpected error (HTTP code $http_code) occured for registry/image $(params.IMAGE_REGISTRY)/$(params.IMAGE_REPOSITORY) during running conftest" > $(workspaces.sanity-ws.path)/stderr.txt
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        fi
+            failure_counter=$((failure_counter+$(jq -r '.[].failures|length' ${IMAGE_REPO_PATH}/deprecated_image_check_output.json)))
+            success_counter=$((success_counter+$(jq -r '.[].successes' ${IMAGE_REPO_PATH}/deprecated_image_check_output.json)))
 
-    # Format the test results and save them as the HACBS_TEST_OUTPUT Tekton result
-    - name: test-format-result
-      image: quay.io/redhat-appstudio/hacbs-test:latest
-      script: |
-        if [ ! -z $(cat $(workspaces.sanity-ws.path)/stderr.txt) ]
-        then
-          echo "The deprecated-image-check test meets the following error:"
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        fi
+          elif [ "$http_code" == "404" ];
+          then
+            echo "Registry/image $(IMAGE_REGISTRY)/$(IMAGE_REPOSITORY) not found in Pyxis" >> $(workspaces.sanity-ws.path)/stderr.txt
+            echo $(workspaces.sanity-ws.path)/stderr.txt
+          else
+            echo "Unexpected error (HTTP code $http_code) occured for registry/image $(IMAGE_REGISTRY)/$(IMAGE_REPOSITORY)" >> $(workspaces.sanity-ws.path)/stderr.txt
```

Comment:
Same as Michal mentioned above, I think that the change should be to switch these to be bash variables instead of Tekton ones, so something like this:
```
$(IMAGE_REGISTRY) -> ${IMAGE_REGISTRY}
```

Reply:
Thanks!